File: backend/src/main/java/com/survey/photo/model/Project.java
package com.survey.photo.model;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("projects")
public class Project {
    @TableId(type = IdType.ASSIGN_UUID)
    private String id;
    private String code;
    private String name;
    private String description;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: admin/src/api/project.ts
import request from '@/utils/request'
import type { Project } from '@/types'

// 获取项目列表
export function getProjects() {
  return request.get<Project[]>('/projects')
}

// 获取项目详情
export function getProject(id: string) {
  return request.get<Project>(`/projects/${id}`)
}

// 创建项目
export function createProject(data: {
  code: string
  name: string
  description?: string
}) {
  return request.post<Project>('/projects', data)
}

// 获取项目二维码
export function getProjectQrcode(id: string) {
  return request.get<string>(`/projects/${id}/qrcode`)
}
